File: after phase gate later edits updates  all tasks completing/grading.js

// Main program
const subjects = [];
const scores = [];
const studentNames = [];
const studentTotalScores = [];
const studentAverageScores = [];
const prompt=require("prompt-sync")();
const teacherName = prompt("Enter Teacher's name:");

const numberOfStudents = parseInt(prompt("How many students do you have?:"));
const numberOfSubjects = parseInt(prompt("How many subjects does each student offer?"));

for (let subjectCounter = 0; subjectCounter < numberOfSubjects; subjectCounter++) {
    const subjectName = prompt("Enter name of subject #number" + (subjectCounter + 1) + ":");
    subjects.push(subjectName);
}

for (let studentCounter = 0; studentCounter < numberOfStudents; studentCounter++) {
    console.log("\nENTERING SCORES FOR STUDENT NUMBER#" + (studentCounter + 1));
    const currentStudentName = prompt("Enter name of student #" + (studentCounter + 1) + ":");
    studentNames.push(currentStudentName);

    let totalScore = 0;

    for (let j = 0; j < numberOfSubjects; j++) {
        const score = parseInt(prompt("Enter score for " + subjects[j] + ":"));
        scores.push(score);
        totalScore += score;
    }

    studentTotalScores.push(totalScore);
    const average = totalScore / numberOfSubjects;
    studentAverageScores.push(average);

    console.log("Saving >>>>>>>>>>>>>>>.");
    console.log("Saved successfully");
}

displayClassSummary(
    teacherName,
    subjects,
    scores,
    studentNames,
    studentTotalScores,
    studentAverageScores,
    numberOfStudents,
    numberOfSubjects
);

function displayClassSummary(
    teacherName,
    subjects,
    scores,
    studentNames,
    studentTotalScores,
    studentAverageScores,
    numberOfStudents,
    numberOfSubjects
) {
    console.log("Teacher Name:", teacherName);
    console.log("Date:", new Date());
    console.log("Time:", new Date().toLocaleTimeString());
    console.log("------");

    // Calculate positions
    const positions = [];
    for (let currentStudentIndex = 0; currentStudentIndex < numberOfStudents; currentStudentIndex++) {
        let position = 1;
        for (let comparisonStudentIndex = 0; comparisonStudentIndex < numberOfStudents; comparisonStudentIndex++) {
            if (studentTotalScores[comparisonStudentIndex] > studentTotalScores[currentStudentIndex]) {
                position++;
            }
        }
        positions.push(position);
    }

    // Display tabular summary
    console.log("\nCLASS SUMMARY");
    let header = "STUDENT\t";
    for (const subject of subjects) {
        header += subject + "\t";
    }
    header += "TOT\tAVE\tPOS";
    console.log(header);

    for (let studentIndex = 0; studentIndex < numberOfStudents; studentIndex++) {
        let row = studentNames[studentIndex] + "\t";
        for (let subjectIndex = 0; subjectIndex < numberOfSubjects; subjectIndex++) {
            const scorePosition = studentIndex * numberOfSubjects + subjectIndex;
            row += scores[scorePosition] + "\t";
        }
        row += studentTotalScores[studentIndex] + "\t" + studentAverageScores[studentIndex].toFixed(2) + "\t" + positions[studentIndex];
        console.log(row);
    }

    // Subject Analysis
    console.log("\nSUBJECT ANALYSIS");
 for (let subjectIndex = 0; subjectIndex < numberOfSubjects; subjectIndex++) {
        const firstScore = scores[subjectIndex];
        let highestScore = firstScore;
        let lowestScore = firstScore;
        let totalScoreForSubject = 0;
        let numberOfPasses = 0;
        let numberOfFails = 0;
        let topScoringStudent = studentNames[0];
        let lowestScoringStudent = studentNames[0];

        for (let studentIndex = 0; studentIndex < numberOfStudents; studentIndex++) {
            const scorePosition = studentIndex * numberOfSubjects + subjectIndex;
            const currentScore = scores[scorePosition];
            totalScoreForSubject += currentScore;

            if (currentScore >= 50) {
                numberOfPasses++;
            } else {
                numberOfFails++;
            }

            if (currentScore > highestScore) {
                highestScore = currentScore;
                topScoringStudent = studentNames[studentIndex];
            }
            if (currentScore < lowestScore) {
                lowestScore = currentScore;
                lowestScoringStudent = studentNames[studentIndex];
            }
        }

        const averageScoreForSubject = totalScoreForSubject / numberOfStudents;
        console.log("Subject:", subjects[subjectIndex]);
        console.log("Highest scoring student:", topScoringStudent, "scoring", highestScore);
        console.log("Lowest scoring student:", lowestScoringStudent, "scoring", lowestScore);
        console.log("Total Score:", totalScoreForSubject);
        console.log("Average Score:", averageScoreForSubject);
        console.log("Passes:", numberOfPasses);
        console.log("Fails:", numberOfFails);
        console.log("---------------------");
    }

    console.log("\nTHANK YOU FOR USING LAGBAJA SCHOOL SYSTEM!");
}

File: after phase gate later edits updates  all tasks completing/grading.py

from datetime import date, datetime

# Main program
subjects = []
scores = []
studentNames = []
studentTotalScores = []
studentAverageScores = []

print("Enter Teachers name:")
teacherName = input()

print("How many students do you have?:")
numberOfStudents = int(input())

print("How many subjects does each student offer?")
numberOfSubjects = int(input())

for i in range(numberOfSubjects):
    print("Enter name of subject #number" + str(i + 1) + ":")
    subjectName = input()
    subjects.append(subjectName)

for i in range(numberOfStudents):
    print("\nENTERING SCORES FOR STUDENT NUMBER#" + str(i + 1))
    print("Enter name of student #" + str(i + 1) + ":")
    currentStudentName = input()
    studentNames.append(currentStudentName)

    totalScore = 0

    for j in range(numberOfSubjects):
        print("Enter score for " + subjects[j] + ":")
        score = int(input())
        scores.append(score)
        totalScore += score

    studentTotalScores.append(totalScore)
    average = totalScore / numberOfSubjects
    studentAverageScores.append(average)

    print("Saving >>>>>>>>>>>>>>>.")
    print("Saved successfully")
displayClassSummary(teacherName, subjects, scores, studentNames, studentTotalScores, studentAverageScores, numberOfStudents, numberOfSubjects)

def displayClassSummary(teacherName, subjects, scores, studentNames, studentTotalScores, studentAverageScores, numberOfStudents, numberOfSubjects):
    print("Teacher Name:", teacherName)
    print("Date:", date.today())
    print("Time:", datetime.now())

    print("------")

    # Calculate positions
    positions = []
    for currentStudentIndex in range(numberOfStudents):
        position = 1
        for comparisonStudentIndex in range(numberOfStudents):
            if studentTotalScores[comparisonStudentIndex] > studentTotalScores[currentStudentIndex]:
                position += 1
        positions.append(position)

    # Display tabular summary
    print("\nCLASS SUMMARY")
    print("STUDENT\t", end="")
    for subject in subjects:
        print(subject + "\t", end="")
    print("TOT\tAVE\tPOS")

    # Display each student's scores and summary
for studentIndex in range(numberOfStudents):
    print(studentNames[studentIndex] + "\t", end="")
    for subjectIndex in range(numberOfSubjects):
        scorePosition = studentIndex * numberOfSubjects + subjectIndex
        print(str(scores[scorePosition]) + "\t", end="")
    print(str(studentTotalScores[studentIndex]) + "\t" + str(studentAverageScores[studentIndex]) + "\t" + str(positions[studentIndex]))

# Subject Analysis
print("\nSUBJECT ANALYSIS")
for subjectIndex in range(numberOfSubjects):

    firstScore = scores[subjectIndex]  

    highestScore = firstScore
    lowestScore = firstScore
  
    totalScoreForSubject = 0
    numberOfPasses = 0
    numberOfFails = 0
    topScoringStudent = studentNames[0]
    lowestScoringStudent = studentNames[0]

    for studentIndex in range(numberOfStudents):
        scorePosition = studentIndex * numberOfSubjects + subjectIndex
        currentScore = scores[scorePosition]
        totalScoreForSubject += currentScore

        if currentScore >= 50:
            numberOfPasses += 1
        else:
            numberOfFails += 1

        if currentScore > highestScore:
            highestScore = currentScore
            topScoringStudent = studentNames[studentIndex]
        if currentScore < lowestScore:
            lowestScore = currentScore
            lowestScoringStudent = studentNames[studentIndex]

    averageScoreForSubject = totalScoreForSubject / numberOfStudents
    print("Subject:", subjects[subjectIndex])
    print("Highest scoring student:", topScoringStudent, "scoring", highestScore)
    print("Lowest scoring student:", lowestScoringStudent, "scoring", lowestScore)
    print("Total Score:", totalScoreForSubject)
    print("Average Score: {:.2f}".format(averageScoreForSubject))
    print("Passes:", numberOfPasses)
    print("Fails:", numberOfFails)
    print("---------------------")

print("\nTHANK YOU FOR USING LAGBAJA SCHOOL SYSTEM!")


